### PR TITLE
feat: Validate schemas in to_allotrope

### PR DESCRIPTION
**Pull request by nathan-stender**
This PR adds validation of the serialized schema in `to_allotrope` methods.

This prevents the library from returning an invalid schema for a file not covered in unit tests.

It also removes tests that only tested schema without testing contents, reducing test run time from ~80s to ~50s

### CODE REVIEW

File: src/allotropy/allotrope/allotrope.py [hunk 1]
```diff
@@ -135,3 +138,28 @@ def make_unstructure_fn(subcls: Any) -> Callable[[Any], dict[str, Any]]:
...
+    try:
+        allotrope_dict = serialize_allotrope(model)
+    except Exception as e:
+        msg = f"Failed to serialize allotrope model: {e}"
+        raise AllotropeConversionError(msg) from e
+
+    try:
+        allotrope_schema = get_schema_from_model(model)
+    except Exception as e:
+        msg = f"Failed to retrive schema for model: {e}"
```
Comment (stephenworlow) [inline src/allotropy/allotrope/allotrope.py:153]:
```suggestion
        msg = f"Failed to retrieve schema for model: {e}"
```

File: src/allotropy/allotrope/schemas.py [hunk 2]
```diff
@@ -25,3 +26,23 @@ def add_definitions(schema: dict[str, Any]) -> dict[str, Any]:
...
+
+
+def get_schema_from_manifest(manifest: str) -> dict[str, Any]:
+    return get_schema(get_schema_path_from_manifest(manifest))
+
+
+def get_schema_from_model(model: Any) -> dict[str, Any]:
+    manifest = getattr(model, "manifest", getattr(model, "field_asm_manifest", None))
+    if not manifest:
+        msg = f"No 'manifest' or 'field_asn_manifest' found in model: {type(model)}"
```
Comment (stephenworlow) [inline src/allotropy/allotrope/schemas.py:46]:
```suggestion
        msg = f"No 'manifest' or 'field_asm_manifest' found in model: {type(model)}"
```

**Review by stephenworlow — APPROVED** [review]

File: src/allotropy/allotrope/allotrope.py (hunk 1 above)
Comment (nathan-stender) [reply src/allotropy/allotrope/allotrope.py:153]: I swear I fixed this 😭

File: src/allotropy/allotrope/schemas.py (hunk 2 above)
Comment (nathan-stender) [reply src/allotropy/allotrope/schemas.py:46]: I swear I... try to spell things correctly 😭 😭

**Review by stephenworlow — APPROVED** [review]

**Comment by bdworth** [comment]
Just saw this since it conflict-bombed my #284 -- nice! Especially the reduction in test run time.

**Pull request merged**